quickwit/quickwit-search/src/leaf.rs

```diff
@@ -394,15 +399,283 @@ async fn leaf_search_single_split(
+        return;
+    };
+
+    let start_timestamp = search_request
+        .start_timestamp
+        .map(DateTime::from_timestamp_secs)
+        .map(Bound::Included)
+        .unwrap_or(Bound::Unbounded);
+    let end_timestamp = search_request
+        .end_timestamp
+        .map(DateTime::from_timestamp_secs)
+        .map(Bound::Excluded)
+        .unwrap_or(Bound::Unbounded);
+
+    let mut visitor = RemoveTimestampRange {
+        timestamp_field,
+        start_timestamp,
+        end_timestamp,
+    };
+    let mut new_ast = visitor
+        .transform(query_ast)
+        .expect("can't fail unwrapping Infallible")
+        .unwrap_or(QueryAst::MatchAll);
+
+    let final_start_timestamp = match (
+        visitor.start_timestamp,
+        split.timestamp_start.map(DateTime::from_timestamp_secs),
+    ) {
+        (Bound::Included(query_ts), Some(split_ts)) => {
+            if query_ts > split_ts {
+                Bound::Included(query_ts)
+            } else {
+                Bound::Unbounded
+            }
+        }
+        (Bound::Excluded(query_ts), Some(split_ts)) => {
+            if query_ts >= split_ts {
+                Bound::Excluded(query_ts)
+            } else {
+                Bound::Unbounded
+            }
+        }
+        (Bound::Unbounded, Some(_)) => Bound::Unbounded,
+        (timestamp, None) => timestamp,
+    };
+    let final_end_timestamp = match (
+        visitor.end_timestamp,
+        split.timestamp_end.map(DateTime::from_timestamp_secs),
+    ) {
+        (Bound::Included(query_ts), Some(split_ts)) => {
+            if query_ts < split_ts {
+                Bound::Included(query_ts)
+            } else {
+                Bound::Unbounded
+            }
+        }
+        (Bound::Excluded(query_ts), Some(split_ts)) => {
+            if query_ts <= split_ts {
+                Bound::Excluded(query_ts)
+            } else {
+                Bound::Unbounded
+            }
+        }
+        (Bound::Unbounded, Some(_)) => Bound::Unbounded,
+        (timestamp, None) => timestamp,
+    };
+    if final_start_timestamp != Bound::Unbounded || final_end_timestamp != Bound::Unbounded {
+        let range = RangeQuery {
+            field: timestamp_field.to_string(),
+            lower_bound: map_bound(final_start_timestamp, |bound| {
+                bound.into_timestamp_nanos().into()
+            }),
+            upper_bound: map_bound(final_end_timestamp, |bound| {
+                bound.into_timestamp_nanos().into()
+            }),
+        };
+        if let QueryAst::Bool(bool_query) = &mut new_ast {
+            bool_query.filter.push(range.into());
+        } else {
+            new_ast = BoolQuery {
+                must: vec![new_ast],
+                filter: vec![range.into()],
+                ..Default::default()
+            }
+            .into();
+        }
+    }
+
+    search_request.query_ast = serde_json::to_string(&new_ast).unwrap();
+    search_request.start_timestamp = None;
```

Comment:
can you check whether this is a regression or not?

If we go that root we should modify the collector too.

Reply:
i've tested 4 scenarii: query with many hits, few hits, with time bounds, and without. Most seems unchanged, time bounds with many hits gets way worse, 11.2ms=>18.7ms. Maybe there is something to reclaim by removing time bounds handling from the collector, but probably not \*that much\*. It looks like it might be a better idea to support higher precision timestamps inside the collector and handle the time bounds there :-/